src/main/ruby/truffleruby/core/array.rb

```diff
@@ -296,11 +296,45 @@ def dig(idx, *more)
     if Primitive.nil?(result) || more.empty?
       result
     else
-      raise TypeError, "#{result.class} does not have #dig method" unless result.respond_to?(:dig)
-      result.dig(*more)
+      Array.looping_dig(result, *more)
     end
   end
 
+  def self.looping_dig(obj, *idxs)
+    idxs.each_with_index do |idx, n|
```

Comment:
Probably needs to be a `while` for better interpreter performance and not consume too much of the inlining budget.

Reply:
Yes that's on our list - and avoiding the `next` exception.